src/nixpacks/builder/docker.rs

```diff
@@ -139,21 +204,25 @@ impl DockerBuilder {
         files::recursive_copy_dir(app_src, &dest)
     }
 
-    fn write_dockerfile(&self, plan: &BuildPlan, dest: &str, env: &Environment) -> Result<()> {
-        let dockerfile = self.create_dockerfile(plan, env);
-
-        let dockerfile_path = PathBuf::from(dest).join(PathBuf::from("Dockerfile"));
-        File::create(dockerfile_path.clone()).context("Creating Dockerfile file")?;
+    fn write_dockerfile(
+        &self,
+        plan: &BuildPlan,
+        dockerfile_path: &str,
+        env: &Environment,
+        environment_nix_path: &str,
```

Comment:
It is a bit strange to be passing in the nix environment path when creating the dockerfile. I think it would be better to just pass in the full path of the dockerfile that should be created.

Reply:
I've removed both & added new `get_nixpacks_file_path` method to resolve absolute path